### ISSUE TITLE
Move Nextflow specific K8s labels under "nextflow.io/" prefix

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#3468]
When submitting the execution of Nextflow tasks some labels are added both the corresponding pods or jobs. See here: 

https://github.com/nextflow-io/nextflow/blob/89e93c02847331fc5f70f042352b256372f699bf/modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy#L278-L282

The goal of this issue is to prefix them with the string `nextflow.io/` to keep under a common scope and prevent conflicts with labels that can the same name.

**Comment by pditommaso** [comment]
@xhejtman @mes5k does sound good to you? could it have any side-effect?

**Comment by xhejtman** [comment]
I think it is ok. But speaking of it, it would be nice to add lables also to pods if using jobs. Currently, lables are on jobs only and it makes it a bit annoying to search among jobs and pods..

**Comment by pditommaso** [comment]
Sure? we always had these labels on the pod spec

https://github.com/nextflow-io/nextflow/blob/89e93c02847331fc5f70f042352b256372f699bf/modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy#L199-L208

**Comment by xhejtman** [comment]
Yes, but if `computeResourceType` is set to Job, labels are on the Job resource only. iirc.

**Comment by pditommaso** [comment]
I see. it would make sense to address into a specific issue then

**Comment by mes5k** [comment]
Are you going to prefix all labels, including those that a user defines for a process or just labels that Nextflow adds automatically?  

I think it's a great idea for labels added by Nextflow automatically.  

It would be a breaking change if you prefixed labels I'm adding myself in processes.  It's certainly not the end of world and easy to fix if this is important, but I'd prefer if you didn't prefix _my_ labels.

**Comment by pditommaso** [comment]
Cool, there's consensus 9951fcd9
